### PR TITLE
Add patroni api client

### PR DESCRIPTION
**Pull request by ikitiki**
currently patroni api client has only one method — failover

### CODE REVIEW

File: pkg/util/patroni/patroni.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+func (p *Patroni) apiURL(masterPod *v1.Pod) string {
+	return fmt.Sprintf("http://%s:%d", masterPod.Status.PodIP, apiPort)
+}
+
+// Failover does manual failover via patroni api
+func (p *Patroni) Failover(master *v1.Pod, candidate string) error {
+	buf := &bytes.Buffer{}
+	err := json.NewEncoder(buf).Encode(map[string]string{"leader": master.Name, "member": candidate})
+
+	request, err := http.NewRequest(http.MethodPost, p.apiURL(master)+failoverPath, buf)
```
Comment (alexeyklyukin) [inline pkg/util/patroni/patroni.go]: a nitpick: I'd add failoverpath to the apiURL method.

Comment (ikitiki) [reply pkg/util/patroni/patroni.go]: what if we want to use other api endpoints?

**Comment by alexeyklyukin** [comment]
LGTM, although I can see 2 corner cases (perhaps they deserve their own PR):

- it could happen that the failover succeeded, but we did not failover to the desired replica (if that replicas was not available), Patroni will return 200 and 'Failed over to "memberA" instead of "memberB"'. 

- it may happen that Patroni returns 503 and "Failover status unknown".

In both cases the caller should recheck the failover results and, possibly, repeat the attempt with another replica.

**Comment by ikitiki** [comment]
👍

**Comment by alexeyklyukin** [comment]
👍

**Pull request merged**